packages/pages/src/vite-plugin/modules/plugin.ts

```diff
@@ -0,0 +1,173 @@
+import { build, createLogger } from "vite";
+import { ProjectStructure } from "../../common/src/project/structure.js";
+import { glob } from "glob";
+import path from "node:path";
+import fs from "node:fs";
+import { convertToPosixPath } from "../../common/src/template/paths.js";
+import { processEnvVariables } from "../../util/processEnvVariables.js";
+import { nodePolyfills } from "vite-plugin-node-polyfills";
+import pc from "picocolors";
+import { addResponseHeadersToConfigYaml } from "../../util/editConfigYaml.js";
+import SourceFileParser, {
+  createTsMorphProject,
+} from "../../common/src/parsers/sourceFileParser.js";
+
+const wrappedCode = (moduleName: string, containerName: string): string => {
+  return `\nconst moduleContainerForBuildUseOnly = document.getElementById('${containerName}');
+if (!moduleContainerForBuildUseOnly) {
+  throw new Error('could not find ${containerName} element');
+}
+
+ReactDOM.render(
+  <React.StrictMode>
+    <${moduleName}/>
+  </React.StrictMode>,
+  moduleContainerForBuildUseOnly
+);
+  `;
+};
+
+const moduleResponseHeader = {
+  pathPattern: "^modules/.*",
+  headerKey: "Access-Control-Allow-Origin",
+  headerValues: ["*"],
+};
+
+export const buildModules = async (
+  projectStructure: ProjectStructure
+): Promise<void> => {
+  if (!shouldBundleModules(projectStructure)) {
+    return;
+  }
+
+  addResponseHeadersToConfigYaml(projectStructure, moduleResponseHeader);
```

Comment:
We'll definitely need to test this manually against a reverse proxy setup, meaning the module is hosted at a RP domain.

Reply:
Done + tested.